### PR TITLE
adding r-ccs genesis app with simple regression test as workload

### PR DESCRIPTION
**Pull request by jdomke**

### CODE REVIEW

**Comment by jdomke** [comment]
linter is complaining about:
```
Run black --diff --check .
All done! ✨ 🍰 ✨
7 files would be left unchanged.
./repo/genesis/application.py:36: equil ==> equal
./repo/genesis/application.py:39: equil ==> equal
./repo/genesis/application.py:42: equil ==> equal
./repo/qws/application.py:63: sovler ==> solver
Error: Process completed with exit code 65.
```
but sadly those are the names of inputs and text in output, so not something i can influence right now

**Comment by pearce8** [comment]
@alecbcs What can be done to force linter to ignore these?

**Comment by jdomke** [comment]
genesis runs now. problem was due to a bug in our llvm.

**Comment by jdomke** [comment]
> @alecbcs What can be done to force linter to ignore these?

i changed the regex slightly to fool the linter, seems a merge is now doable, please go ahead @pearce8

**Review by pearce8 — APPROVED** [review]

**Comment by pearce8** [comment]
@scheibelp Please take a look, I hope you recall this error from before - and how we would fix it:

==> Error: /home/runner/work/benchpark/benchpark/workspace/genesis/openmp/RCCS-Fugaku-Fujitsu-A64FX-TofuD/workspace/configs/ramble.yaml:39
    concretized: true

Additional properties are not allowed ('concretized' was unexpected)

File: experiments/genesis/openmp/ramble.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            experiment_setup: ''
+            processes_per_node: ['{sys_cores_per_node} / 6']
+            n_nodes: ['1']
+            omp_num_threads: ['{processes_per_node} / 2']
+            arch: 'OpenMP'
+          experiments:
+            genesis_omp_{n_nodes}_{processes_per_node}_{omp_num_threads}:
+              variables: {}
+  spack:
+    concretized: true
```
Comment (scheibelp) [inline experiments/genesis/openmp/ramble.yaml]: I think you just need to remove `concretized: true` for this to pass the tests: see https://github.com/LLNL/benchpark/pull/265#pullrequestreview-2105221682

**Comment by scheibelp** [comment]
The new error indicates to me you may need to merge latest benchpark into your branch: 

* https://github.com/LLNL/benchpark/pull/301  changed `spack.yaml` to `software.yaml` based on the new Ramble
* If the CI is based on an older Benchpark (using an older Ramble) that would explain why it's looking for (and failing to find) `spack.yaml`

**Comment by jdomke** [comment]
> The new error indicates to me you may need to merge latest benchpark into your branch:
> 
>     * [Update benchpark for ramble package manager abstraction #301](https://github.com/LLNL/benchpark/pull/301)  changed `spack.yaml` to `software.yaml` based on the new Ramble
> 
>     * If the CI is based on an older Benchpark (using an older Ramble) that would explain why it's looking for (and failing to find) `spack.yaml`

fixed it

**Comment by jdomke** [comment]
> > The new error indicates to me you may need to merge latest benchpark into your branch:
> > ```
> > * [Update benchpark for ramble package manager abstraction #301](https://github.com/LLNL/benchpark/pull/301)  changed `spack.yaml` to `software.yaml` based on the new Ramble
> > 
> > * If the CI is based on an older Benchpark (using an older Ramble) that would explain why it's looking for (and failing to find) `spack.yaml`
> > ```
> 
> fixed it

ok, at least i tried, can you make sense of the current error @scheibelp ?

File: experiments/genesis/openmp/ramble.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# Copyright 2023 Lawrence Livermore National Security, LLC and other
+# Benchpark Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+ramble:
+  include:
+  - ./configs/spack.yaml
```
Comment (scheibelp) [inline experiments/genesis/openmp/ramble.yaml]: To get the `--dry-run` to work, I had to make the following changes:

```
diff --git a/configs/RCCS-Fugaku-Fujitsu-A64FX-TofuD/software.yaml b/configs/RCCS-Fugaku-Fujitsu-A64FX-TofuD/software.yaml
index ab7f49c..69a165e 100644
--- a/configs/RCCS-Fugaku-Fujitsu-A64FX-TofuD/software.yaml
+++ b/configs/RCCS-Fugaku-Fujitsu-A64FX-TofuD/software.yaml
@@ -6,15 +6,18 @@
 software:
   packages:
     default-compiler:
-      pkg_spec: "{default_comp} {sys_arch}"
+      pkg_spec: "fj"
     default-mpi:
-      pkg_spec: fujitsu-mpi@{fj_comp_version}%{default_comp} {sys_arch}
+      pkg_spec: "fujitsu-mpi@{default_fj_version}"
+      compiler: default-compiler
     blas:
-      pkg_spec: fujitsu-ssl2@{fj_comp_version}%{default_comp} {sys_arch}
+      pkg_spec: "fujitsu-ssl2@{default_fj_version}"
+      compiler: default-compiler
     lapack:
-      pkg_spec: fujitsu-ssl2@{fj_comp_version}%{default_comp} {sys_arch}
+      pkg_spec: "fujitsu-ssl2@{default_fj_version}"
+      compiler: default-compiler
     gmake:
-      pkg_spec: gmake@4.2.1 {sys_arch}
+      pkg_spec: "gmake@4.2.1"
     cmake:
-      pkg_spec: cmake@3.20.2 {sys_arch}
+      pkg_spec: "cmake@3.20.2"
 
diff --git a/experiments/genesis/openmp/ramble.yaml b/experiments/genesis/openmp/ramble.yaml
index e928b2e..471e530 100644
--- a/experiments/genesis/openmp/ramble.yaml
+++ b/experiments/genesis/openmp/ramble.yaml
@@ -5,7 +5,7 @@
 
 ramble:
   include:
-  - ./configs/spack.yaml
+  - ./configs/software.yaml
   - ./configs/variables.yaml
   - ./configs/modifier.yaml
 
@@ -25,20 +25,22 @@ ramble:
           env_vars:
             set:
               OMP_NUM_THREADS: '{omp_num_threads}'
+          variants:
+            package_manager: spack
           variables:
             n_ranks: '{processes_per_node} * {n_nodes}'
             experiment_setup: ''
-            processes_per_node: ['{sys_cores_per_node} / 6']
+            processes_per_node: ['int({sys_cores_per_node} / 6)']
             n_nodes: ['1']
             omp_num_threads: ['{processes_per_node} / 2']
             arch: 'OpenMP'
           experiments:
             genesis_omp_{n_nodes}_{processes_per_node}_{omp_num_threads}:
               variables: {}
-  spack:
+  software:
     packages:
       genesis:
-        spack_spec: genesis@2.1.3 +mpi+openmp{modifier_spack_variant}
+        pkg_spec: genesis@2.1.3 +mpi+openmp{modifier_spack_variant}
         compiler: default-compiler
     environments:
       genesis:
```

In particular:

- cast calculation of `processes_per_node` to an `int`
- the reference on this line needed correction
- at that point I got errors about `fj_comp_version` being undefined: I didn't see it so I updated the reference to `default_fj_version`

Comment (jdomke) [reply experiments/genesis/openmp/ramble.yaml]: thanks for the suggestions, i fix the issues and a few more things which came up during testing with the new version

**Review by scheibelp — APPROVED** [review]

**Comment by scheibelp** [comment]
Thanks!

**Pull request merged**